packages/shared/src/hooks/useFeedLayout.ts

```diff
@@ -26,18 +29,24 @@ export const useFeedLayout = ({
   feedName: feedNameProp,
   feedRelated = true,
 }: UseFeedLayoutProps = {}): UseFeedLayout => {
+  const hasCommentFeed = useFeature(feature.commentFeed);
   const { feedName } = useActiveFeedNameContext();
   const isMobile = useViewSize(ViewSize.MobileL);
   const name = (feedNameProp ?? feedName) as SharedFeedPage;
   const isIncludedFeed = useMemo(
     () => checkShouldUseMobileFeedLayout(name),
     [name],
   );
-  const shouldUseMobileFeedLayout = !feedRelated
-    ? isMobile
-    : isMobile && isIncludedFeed;
+
+  const shouldUseCommentFeedLayout =
+    hasCommentFeed && feedName === SharedFeedPage.Discussed;
+  const isMobileAndIncludedFeed = isMobile && isIncludedFeed;
+  const shouldUseMobileFeedLayout = feedRelated
+    ? isMobileAndIncludedFeed
+    : isMobile;
```

Comment:
ah, I see now why the `MainFeedLayout` was modified, you're switching to the comments feed on the `/discussed` page if the feature flag is on 👌